libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -363,10 +363,11 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
     }
 
     // when landing, add half of head-wind.
-    const int32_t head_wind_compensation_cm = head_wind() * 0.5f * 100;
+    const float head_wind_comp = constrain_float((float)wind_comp, 0.0f, 100.0f)*0.01;
```

Comment:
Why is this being crazy to a float when it's already a float? Maybe it needs a .get() ? Also, why is it scaled down and then scaled up?

Reply:
* the float cast was templated from :https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Landing/AP_Landing_Slope.cpp#L294
* The scale up and then scale down was more for mathematical clarity and less for pure succinctness.  The first scale down is the transition from a 0-100% to numerical 0-1 equivalent.  The scale up, is the conversion to cm.  It just so happens that 0.01 * 100 ends up being 1.  I'm not opposed to removing it, but could make follow on looks at the code unclear as to how units are being produced by those two lines of code.

Reply:
that landing_slope example you point to is casting to float because the variable it's using is AP_Int8 which is an int8 object accessor, not an actual variable.

Reply:
fixed this